core/domain/opportunity_services.py

```diff
@@ -384,3 +403,32 @@ def update_opportunities_with_new_topic_name(topic_id, topic_name):
 
     _save_multi_exploration_opportunity_summary(
         exploration_opportunity_summary_list)
+
+
+def regenerate_exploration_opportunities_summary():
+    """"Regenerates all of the exploration opportunity summary models."""
+    opportunity_models.ExplorationOpportunitySummaryModel.delete_all()
+
+    story_ids = []
+    topics = topic_fetchers.get_all_topics()
+    for topic in topics:
+        story_ids += topic.get_canonical_story_ids()
```

Comment:
Should I use a set here? (though we don't expect to have a duplicate value, also having duplicate value won't cause any issue.)